src/s3poller.ts

```diff
@@ -80,14 +82,32 @@ export async function pollS3Source(
 
     // Handle case where there are common prefixes (folder structure)
     if (prefixes && prefixes.length > 0) {
-      const batchSize = 20
+      const batchSize = 100
       for (let i = 0; i < prefixes.length; i += batchSize) {
         const batch = prefixes.slice(i, i + batchSize)
+        
+        // Collect all files from all prefixes in this batch
+        const allFiles: any[] = []
+        
+        // Fetch files from all prefixes in parallel for efficiency
         await Promise.all(
           batch.map(async (prefix) => {
-            await scanS3Prefix(sourceName, client, bucket, prefix)
+            const prefixResult = await client.send(
+              new ListObjectsCommand({
+                Bucket: bucket,
+                Prefix: prefix,
+              })
+            )
+            if (prefixResult.Contents) {
+              allFiles.push(...prefixResult.Contents)
+            }
           })
         )
+
+        // Process all files in chronological order
```

Comment:
I will admit I was at first confused as to how this is ensuring chronological ordering haha. But I see that's in the implementation details of `processS3Contents`